exercises/bowling/example.rb

```diff
@@ -0,0 +1,66 @@
+class Game
+  class InvalidRoll < StandardError; end
+  class GameInProgress < StandardError; end
+  class GameHasEnded < StandardError; end
+
+  VERSION = 1
+  TOTAL_FRAMES = 10
+  MINIMUM_ROLL_SCORE = 0
+  MAXIMUM_ROLL_SCORE = 10
+
+  def initialize
+    @frame_scores = []
+    @frame_buffer = []
+  end
+
+  def roll(pins)
+    validate_roll(pins)
+    @frame_buffer << pins
+    adjust_frames
+  end
+
+  def score
+    raise GameInProgress unless @frame_scores.size == TOTAL_FRAMES
+    @frame_scores.reduce(:+)
+  end
+
+  private
+
+  def validate_roll(pins)
+    raise InvalidRoll if pins < MINIMUM_ROLL_SCORE
+    raise InvalidRoll if pins > MAXIMUM_ROLL_SCORE
```

Comment:
I like the use of `fail` here, but I prefer the `if` statements instead of the `.cover?`. If you really think this reads better, I have no problem in changing this. :)